sapi/cli/php_cli_server.c

```diff
@@ -1639,17 +1642,34 @@ static int php_cli_server_client_read_request_on_path(php_http_parser *parser, c
 static int php_cli_server_client_read_request_on_query_string(php_http_parser *parser, const char *at, size_t length)
 {
 	php_cli_server_client *client = parser->data;
-	client->request.query_string = pestrndup(at, length, 1);
-	client->request.query_string_len = length;
+	if (client->request.query_string == NULL) {
```

Comment:
I think you may be able to drop the query_string == NULL code path -- it's fine to call perealloc with NULL. Feel free to keep as-is though.

Reply:
I find that code path way more readable, and since the other leg isn't to be expected, I would prefer to keep as is, but I'm going to put in `EXPECTED()`  for clarity and optimization.